core/client/fs/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -641,19 +829,16 @@ private long remainingInternal() {
    */
   private void seekInternalWithCachingPartiallyReadBlock(long pos) throws IOException {
     // Precompute this because mPos will be updated several times in this function.
-    final boolean isInCurrentBlock = pos / mBlockSize == mPos / mBlockSize;
+    final boolean isInCurrentBlock = mCurrentBlockInStream != null
+        && mPositionState.isInCurrentBlock(pos);
 
     if (isInCurrentBlock) {
-      // the read is within the current block update stream to refresh the streams
-      updateStreams();
       if (isReadFromLocalWorker()) {
         // no need to partial cache the current block, and the seek is within the block
         // so directly seeks to position.
-        mPos = pos;
-        // updateStreams is necessary when pos = mFileLength.
-        updateStreams();
+        mPositionState.setPos(pos);
         if (mCurrentBlockInStream != null) {
-          mCurrentBlockInStream.seek(mPos % mBlockSize);
+          mCurrentBlockInStream.seek(mPositionState.getPos() % mBlockSize);
```

Comment:
if the seek position is in the first block for the first read of the opened file, without `updateStreams()` the `mCurrentBlockInStream`  would not be initialized and therefore be null, right?

Reply:
yes

Reply:
but this sounds like a bug to me, because the else condition should fail: `remaining() == 0`

Reply:
if `isInCurrentBlock` is `true`, then the `else` is unnecessary, I've updated the PR to remove the condition check here.